http-gateway/service/config.go

```diff
@@ -114,36 +114,63 @@ func (c *ClientsConfig) Validate() error {
 	return nil
 }
 
-// OAuthClientConfig represents oauth configuration for user interface exposed via getOAuthConfiguration handler
-type OAuthClientConfig struct {
-	Domain             string `json:"domain" yaml:"domain"`
-	ClientID           string `json:"clientID" yaml:"clientID"`
-	Audience           string `json:"audience" yaml:"audience"`
-	Scope              string `json:"scope" yaml:"scope"`
-	HTTPGatewayAddress string `json:"httpGatewayAddress" yaml:"httpGatewayAddress"`
+type BasicOAuthClient struct {
+	ClientID string   `json:"clientID" yaml:"clientID"`
+	Audience string   `json:"audience" yaml:"audience"`
+	Scopes   []string `json:"scopes" yaml:"scopes"`
 }
 
-func (c *OAuthClientConfig) Validate() error {
-	if c.Domain == "" {
-		return fmt.Errorf("domain('%v')", c.Domain)
-	}
+func (c *BasicOAuthClient) Validate() error {
 	if c.ClientID == "" {
-		return fmt.Errorf("clientID('%v')", c.ClientID)
+		return fmt.Errorf("domain('%v')", c.ClientID)
 	}
 	if c.Audience == "" {
 		return fmt.Errorf("audience('%v')", c.Audience)
 	}
+	return nil
+}
+
+type DeviceOAuthClient struct {
+	BasicOAuthClient `yaml:",inline"`
+	ProviderName     string `json:"providerName" yaml:"providerName"`
+}
+
+func (c *DeviceOAuthClient) Validate() error {
+	if c.ClientID == "" {
+		return fmt.Errorf("domain('%v')", c.ClientID)
+	}
+	return nil
```

Comment:
`return BasicOAuthClient.Validate()`

Reply:
for device provider audience is optional, so this is why DeviceOAuthClient has owned the `Validate` function.